### DISCUSSION TITLE
Question about the download url for the runtime installer

### DISCUSSION THREAD

**Discussion by simen-wu** [MicrosoftEdge/WebView2Feedback#3996] [Q&A]
From WebView2's homepage, I found the download url for the installer is as below:
https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/afcf8876-9ac9-472f-8b64-84548c712806/MicrosoftEdgeWebView2RuntimeInstallerX64.exe

Is this url tied to the current version (106.0.1370.47)? For example, in the future when a newer version is available, will the url be different?

I'm creating a WiX bundle and would like to include WebView2 installer as a remote payload (user downloads the software during the installation). So I want to make sure this particular download url is fixed for version 106.0.1370.47.

**Comment by nishitha-burman** [comment]
Hi @simen-wu, 

Where in the page did you find this link? You can select which version you want to download from this page: https://developer.microsoft.com/en-us/microsoft-edge/webview2/

<img width="212" alt="image" src="https://user-images.githubusercontent.com/54861497/196508542-700de2d4-9b05-41d9-bf39-fd0d7f488930.png">

**Comment by simen-wu** [comment]
@nishitha-burman Thank you for your quick reply. The screenshot you show is for the download of *.cab file. But I'm interested in the standalone installer. Please see this screenshot:

![image](https://user-images.githubusercontent.com/33633190/196518871-7b87472f-69c2-4443-89ea-b1e8bc1f7f4b.png)

**Comment by sln162** [comment]
@simen-wu ![image](https://user-images.githubusercontent.com/11196986/197233317-b0a8023b-c1b8-40d2-a991-b797571a2886.png)
[You can use this](https://go.microsoft.com/fwlink/?linkid=2124701)

**Comment by simen-wu** [comment]
Thanks @sln162. Yes, I can get the actual url by inspecting the html page. But I would like somebody to confirm that the URL you showed is only for version 106.0.1370.47 specifically. 

We are going to include WebView2 runtime in our installer and let user download the runtime from the URL (as you showed) during the installation. During the download, there are some verifications e.g. file version, hash etc. Let's say, in the future, if WebView2 is upgraded to 107.* and the URL is the same, then the verification will fail because file version and hash is different from now (when our installer is designed). And our installer will break.

**Comment by sln162** [comment]
@simen-wu It happened that I downloaded several times in the last month. Through comparison, I found that the hash of urls and files are different, but the version number of the EXE has not changed. I am not sure whether the version numbers of WebView2 are consistent. But I think it is not reliable to judge the version only from the hash of the URL.
![image](https://user-images.githubusercontent.com/11196986/197240853-e88b3d59-e78c-438c-89d4-7f4535629c72.png)
The hashes in the URLs have changed for the four times, but the version of WebView2 has certainly not changed for the four times. You can also download the files for the four times and install them to confirm.
https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/9c4c4193-a32c-4083-92c7-2e49bac0b904/MicrosoftEdgeWebView2RuntimeInstallerX64.exe
https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/0da43a99-5135-4303-adb4-eae68e11969f/MicrosoftEdgeWebView2RuntimeInstallerX64.exe
https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/143fd123-194c-464e-97ab-1d88afa99cbf/MicrosoftEdgeWebView2RuntimeInstallerX64.exe
https://msedge.sf.dl.delivery.mp.microsoft.com/filestreamingservice/files/592b44fb-644a-48a1-9d98-a5d4c1f0a724/MicrosoftEdgeWebView2RuntimeInstallerX64.exe
